keps/sig-testing/20190118-breaking-apart-the-kubernetes-test-tarball.md

```diff
@@ -0,0 +1,274 @@
+title: Breaking apart the Kubernetes test tarball
+authors:
+  - "@ixdy"
+owning-sig: sig-testing
+participating-sigs:
+  - sig-release
+reviewers:
+  - TBD
+approvers:
+  - "@spiffxp"
+editor: TBD
+creation-date: 2019-01-18
+last-updated: 2019-01-18
+status: provisional
+---
+
+# Breaking apart the kubernetes test tarball
+
+## Table of Contents
+
+- [Breaking apart the kubernetes test tarball](#breaking-apart-the-kubernetes-test-tarball)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Internal structure of the test tarball](#internal-structure-of-the-test-tarball)
+      - [Binary artifacts](#binary-artifacts)
+      - [Portable sources](#portable-sources)
+    - [Updating dependencies on `kubernetes-test.tar.gz`](#updating-dependencies-on-kubernetes-testtargz)
+      - [Dependencies outside the Kubernetes organization](#dependencies-outside-the-kubernetes-organization)
+    - [Risks and Mitigations](#risks-and-mitigations)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [References](#references)
+  - [Implementation History](#implementation-history)
+
+## Release Signoff Checklist
+
+- [x] [k/enhancements issue in release milestone and linked to KEP](https://github.com/kubernetes/enhancements/issues/714)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+The Kubernetes release artifacts include a "mondo" test tarball which includes both
+"portable" test sources (such as shell scripts and image manifests) as well as
+platform-specific test binaries for all supported client, node, and
+server platforms.
+
+This KEP proposes replacing this monolith test tarball with many platform-specific
+tarballs, matching the existing pattern used for the client, node, and server
+tarballs.
+
+## Motivation
+
+As the number of supported client, server, and node platforms has increased, the
+size of the test tarball has increased as well. As of Kubernetes v1.13.2, the
+official `kubernetes-test.tar.gz` is approximately 1.2GB; previous releases
+ranged from 1.3 - 1.5GB. Several years ago,
+[there were complaints](https://github.com/kubernetes/kubernetes/issues/28435)
+that the "full" `kubernetes.tar.gz` release tarball was too big at 1.4G.
+Many of the motivations for breaking up that tarball echo into this proposal.
+
+Another motivation comes from the Bazel effort. It's possible to build all
```

Comment:
Hi @ixdy,

`s/Another motivation comes from the Bazel effort/The Bazel effort is another driving motivation./`

Reply:
Hi @ixdy,

I feel like the two sentences beginning with `There is no support` could be restated a little more succinctly? Maybe?

Reply:
I've also since learned that this sentence is not really correct; Bazel may even have this ability today, though it's poorly documented and we're not using it (yet).